spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultSchedulerService.java

```diff
@@ -0,0 +1,183 @@
+		// runner and executable graph.
+		if (taskNode.isComposed()) {
+			taskDefinition = new TaskDefinition(taskDefinition.getName(),
+					TaskServiceUtils.createComposedTaskDefinition(
+							taskNode.toExecutableDSL(), this.taskConfigurationProperties));
+			taskDeploymentProperties = TaskServiceUtils.establishComposedTaskProperties(taskDeploymentProperties, taskNode);
+		}
+
+		AppRegistration appRegistration = this.registry.find(taskDefinition.getRegisteredAppName(),
+				ApplicationType.task);
+		Assert.notNull(appRegistration, "Unknown task app: " + taskDefinition.getRegisteredAppName());
+		Resource metadataResource = this.registry.getAppMetadataResource(appRegistration);
+
+		taskDefinition = TaskServiceUtils.updateTaskProperties(taskDefinition, this.dataSourceProperties);
+
+		Map<String, String> appDeploymentProperties = new HashMap<>(commonApplicationProperties.getTask());
+		appDeploymentProperties.putAll(
+				TaskServiceUtils.extractAppProperties(taskDefinition.getRegisteredAppName(), taskDeploymentProperties));
+
+		Map<String, String> deployerDeploymentProperties = DeploymentPropertiesUtils
+				.extractAndQualifyDeployerProperties(taskDeploymentProperties, taskDefinition.getRegisteredAppName());
+		if (StringUtils.hasText(this.dataflowServerUri) && taskNode.isComposed()) {
+			updateDataFlowUriIfNeeded(appDeploymentProperties, commandLineArgs);
+		}
+		AppDefinition revisedDefinition = TaskServiceUtils.mergeAndExpandAppProperties(taskDefinition, metadataResource,
+				appDeploymentProperties, whitelistProperties);
+		ScheduleRequest scheduleRequest = new ScheduleRequest(revisedDefinition,
+				TaskServiceUtils.extractSchedulerProperties(taskDefinition.getRegisteredAppName(), taskDeploymentProperties),
+				deployerDeploymentProperties, scheduleName, getTaskResource(taskDefinitionName));
+		this.scheduler.schedule(scheduleRequest);
+	}
+
+	@Override
+	public void unschedule(String scheduleName) {
+		this.scheduler.unschedule(scheduleName);
+	}
+
+	@Override
+	public List<ScheduleInfo> list(Pageable pageable, String taskDefinitionName) {
+		// Need to add support for pagination
+		return scheduler.list(taskDefinitionName);
+	}
+
+	@Override
+	public List<ScheduleInfo> list(Pageable pageable) {
+		// Need to add support for pagination
+		return scheduler.list();
+	}
+
+	private Resource getTaskResource(String taskDefinitionName ) {
+		TaskDefinition taskDefinition = this.taskDefinitionRepository.findOne(taskDefinitionName);
+		if (taskDefinition == null) {
+			throw new NoSuchTaskDefinitionException(taskDefinitionName);
+		}
+
+		AppRegistration appRegistration = this.registry.find(taskDefinition.getRegisteredAppName(),
+				ApplicationType.task);
+		Assert.notNull(appRegistration, "Unknown task app: " + taskDefinition.getRegisteredAppName());
+		return this.registry.getAppResource(appRegistration);
+	}
+
+	private void updateDataFlowUriIfNeeded(Map<String, String> appDeploymentProperties, List<String> commandLineArgs) {
```

Comment:
Can we move some of these common methods (here and in DefaultTaskService) into a separate utility class?

Reply:
I see we have one. This can be moved into TaskServiceUtils